ic-os/components/misc/config/setupos/config.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# Shared config utilities.
+
+# Retrieves a value from the config.json file using a JSON path.
+# Arguments:
+#   $1 - JSON path to the desired value (e.g., '.icos_settings.nns_urls')
+# Note: If the key is not found, this function will return null.
+function get_config_value() {
```

Comment:
I remember already seeing a function for reading config values in a previous PR (but can't find it anymore), is there a difference?

Reply:
Unfortunately, yes, we need a config.sh file for SetupOS and another for HostOS/GuestOS. See the *full* configuration PR with both config.sh files: https://github.com/dfinity/ic/pull/1563/files#diff-8760e39edecd962b0efb7167e82feeaa695bcf45af62f0dc1fcdd01a35b058a0

The only difference is the path of the `CONFIG_FILE`. SetupOS's path is `/var/ic/config/config.json` because the config object is generated at runtime, and SetupOS boots as readonly, so its config object is written to var.